### PR TITLE
Feat/guidelines

### PR DESCRIPTION
**Pull request by BenjaSanchez**
### Main improvements in this PR:
* Added code of conduct
* Added contribution guidelines, including guidelines for:
  * External contributors
  * The development team
  * The administrator of the repo

Hopefully I covered most of the "how-to's", if something is missing do let me know! @feiranl @hongzhonglu @ChristianLieven @demilappa @Hao-Chalmers

**I hereby confirm that I have:**
- [x] Selected `devel` as a target branch (top left drop-down menu)
- [x] If needed, asked first in the [Gitter chat room](https://gitter.im/SysBioChalmers/yeast-GEM) about this PR

### CODE REVIEW

**Comment by mihai-sysbio** [comment]
Just a small comment: my feeling is that the barrier of entry to contributing is, at the moment, higher than optimal. It would be great if it could be simplified, e.g. demoed.

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+    * If your changes are minor (e.g. a single chemical formula you wish to correct), you can do it directly from the command line.
+    * If your changes are not so small and require several steps, create a script that loads the model, reads data (if applicable), changes the model accordingly, and saves the model back.
+    * Each script should start with a commented section describing the script, explaining the parameters, and indicating your name and the date it was written. Existing functions can clarify what style should be used.
+    * Store scripts in the appropriate folder in `/ComplementaryScripts` and data (as `.tsv` files) in the appropriate folder in `/ComplementaryData`. If you think no folder is adequate for your script/data, feel free to create your own folder. Note that binary data such as `.mat` structures or `.xls` tables cannot be stored in the repo (as they cannot be version-controlled, and they increment too much the size of the repo).
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: You can also just add common binary file formats to the .gitignore, such as `.mat` and similar Matlab files, and `.xlsx` and other Office files. This wouldn't require a text change here though.

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    * When you are done making changes, review locally your changes with `git diff` or any git client, to make sure you are doing to the model what you intended.
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]:
```suggestion
    * When you are done making changes, review locally your changes with `git diff` or any git client, to make sure you are modifying the model as you intended.
```

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+
+* `{chore, doc, feat, fix, refactor, style}/descriptive-name`: Any other branch created in the model. If you work on a fix, start the branch name with `fix/`, if you work on a feature, start the branch name with `feat/`. Examples: `fix/format_reactions` or `feat/new_algorithms`. [See below](#semantic-commits) for more details on the possible actions you can use.
+	
+#### Semantic commits
+
+Please use concise descriptive commit messages. Ideally, use semantic commit messages to make it easier to show what you are aiming to do:
+
+`action-object: brief description`
+* `action` = {`feat`, `fix`, `refactor`, `style`, `doc`, `chore`, `test`}
+* `object` = {`rxn`, `rxn.annot`, `rxn.prop`, `met`, `met.annot`, `met.prop`, `gene`, `gene.annot`, `comp`, `comp.annot`, `data`}
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: I don't see that much value in the `object` part, you'd probably want to read the full commit message anyway, and to me this seems to make things more complex. But okay to keep it like this.

Comment (edkerk) [reply .github/CONTRIBUTING.md]: Below it is actually mentioned that `object` is optional, maybe mention that here already as well.

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+|Add metabolite formula|`feat-met.prop: carbohydrate formulas`|
+|Fix rxn stoichiometry|`fix-rxn.prop: complex V stoich coeffs`|
+|Update gene IDs|`fix-gene.annot: update IDs from SWISSPROT`|
+|Format name of compartment|`style-comp.annot: remove uppercases`|
+|Split a rxn in 2|`refactor-rxn: split isomerase in 2 steps`|
+|Add some data|`feat-data: metabolomics data`|
+|Update documentation of function|`doc: addDBnewRxn.m`|
+|Update toolbox|`chore: update RAVEN version`|
+
+More examples [here](https://github.com/SysBioChalmers/yeast-GEM/commits/master). A more detailed explanation or comments can be left in the commit description.
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: Shouldn't more detailed explanations be encouraged?

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+* **Documentation:** Every change should be justified with a reference/link/argument. This can be provided as data in `/ComplementaryData`, or directly as a comment in the pull request.
+* **Reproducibility:** If there are any added scripts, make sure that if you run them, the model gets updated from how it was in `devel` to how it is in the pull request. For this, you may _locally_ switch to the corresponding branch, replace the `.xml` file with the one from `devel` before the changes, and run the associated scripts. Remember to stash any undesired changes afterwards.
+* **Style:** Ensure that the changes to the model are compliant with the model's rxn/met/gene naming conventions (when unsure, take a look at a similar field in the model). Also, make sure that scripts have a compliant style, and datasets are straight-forward to understand.
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: Should the naming convention (especially IDs) be specified somewhere? Can IDs be reused for instance?

File: .github/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+* When commenting in the review, please comply with our [code of conduct](https://github.com/SysBioChalmers/yeast-GEM/blob/master/.github/CODE_OF_CONDUCT.md).
+* Avoid vague comments and try to be as explicit as possible (e.g.: _"please change this"_ instead of _"this could be changed"_).
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: To be honest, the examples that you give here seem vague..

File: .github/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+  * Fixing some chemical formulas/charges.
+  * Updating toolboxes.
+  * Re-organization of data
+  * Refactoring of code.
+  
+When releasing, please follow these steps:
+  1. Make sure all dependencies in `devel` correspond to the setup from the local computer from which the release will be made. If not, make a single commit in `devel` updating this with a `loadYeastModel`/`saveYeastModel` cycle.
+  2. Create a pull request from `devel` to `master`, indicating all new features/fixes/etc. and referencing every previous pull request included. Tip: if any [issue](https://github.com/SysBioChalmers/yeast-GEM/issues) gets solved in the release, write in the pull request description "Fixes #X", where "X" is the issue number. That way the issue will be automatically closed after merge.
+  3. Merge at least a day after (having at least one accepted review).
+  4. Switch locally to `master` and update `history.md`, by putting at the top a description with what is included in the new version.
```
Comment (edkerk) [inline .github/CONTRIBUTING.md]: Should new entries in `history.md` not have the same text as the PR to master?

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (mihai-sysbio) [reply .github/CONTRIBUTING.md]: Good idea, although one can always `git add --force`

Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: @edkerk this is actually already implemented, hence the wording "cannot"
@mihai-sysbio was not aware of `git add --force`, hopefully future PRs don't do it :)

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: done

File: .github/CONTRIBUTING.md (hunk 4 above)
Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: done

File: .github/CONTRIBUTING.md (hunk 5 above)
Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: It should definitively be specified. However, I think that requires a full section and would rather take it on a new PR with more discussion. In the meantime, I have included a mention of `getNewIndex.m` for generating new met/rxn ids in step 6 of "Contributing to the model" (and I will soon work on documenting said function).

File: .github/CONTRIBUTING.md (hunk 6 above)
Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: these were not intended to be the whole messages, and I have clarified that in the text

File: .github/CONTRIBUTING.md (hunk 7 above)
Comment (BenjaSanchez) [reply .github/CONTRIBUTING.md]: yes it should actually be the same text, I have clarified that now

**Comment by edkerk** [comment]
Also, is `.github` the best location for these documents, instead of the repo root?

**Comment by Hao-Chalmers** [comment]
It appears that this `CODE_OF_CONDUCT.md` suits nearly all repos, and may be just presented/curated as one piece in the GitHub [SysBioHomepage](https://github.com/SysBioChalmers)? Otherwise, there would be ~50 identical copies deposited to every repo. @mihai-sysbio 

Although this PR is closed, maybe still the valid place for this question.

**Comment by BenjaSanchez** [comment]
@Hao-Chalmers agree with this, a general code of conduct would be great. This one can stay for now, and if a general one is provided we can just redirect here to that one.

**Pull request merged**